msal4j-sdk/src/integrationtest/java/com.microsoft.aad.msal4j/TokenCacheIT.java

```diff
@@ -195,6 +197,68 @@ void retrieveAccounts_ADFSOnPrem() throws Exception {
         assertEquals(pca.getAccounts().join().size(), 1);
     }
 
+    @Test
+    void testStaticCache() throws Exception {
+        AppCredentialProvider appProvider = new AppCredentialProvider(AzureEnvironment.AZURE);
+        final String clientId = appProvider.getLabVaultAppId();
+        final String password = appProvider.getLabVaultPassword();
+        IClientCredential credential = ClientCredentialFactory.createFromSecret(password);
+
+        //Create three client applications: one that uses its own instance of a TokenCache,
+        // and two that use the useSharedCache option to use the same static TokenCache
+        ConfidentialClientApplication cca_notStatic = ConfidentialClientApplication.builder(
+                        clientId, credential).
+                authority(TestConstants.MICROSOFT_AUTHORITY).
+                build();
+
+        ConfidentialClientApplication cca_sharedCache1 = ConfidentialClientApplication.builder(
+                        clientId, credential).
+                authority(TestConstants.MICROSOFT_AUTHORITY).
+                useSharedCache(true).
```

Comment:
+ @localden for the public API aspect. 

Currently, the Java API for specifying the cache is https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/3c00622f7af638dcae2283709a02315db2baac41/msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/AccountTest.java#L89

In MSAL.NET we went with a property object, so as to be able to add more cache options not just this boolean. Arguably, we never added anything else :)

Reply:
Yeah, I talked with @neha-bhargava about that [CacheOptions](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/AppConfig/CacheOptions.cs), but I didn't add it because:
- Shared cache was the only option in there after ~2 years, and if there are still no plans to add any other options then it didn't seem like expandability is more important than simplicity
- Shared cache is a pretty big thing to enable, so even if there are more cache options in the future then having this useSharedCache API alongside a CacheOptions class wouldn't be too confusing for app developers